### PR TITLE
SIMSBIOHUB-519: Event schema

### PR DESCRIPTION
**Pull request by al-rosenthal**
## Links to Jira Tickets

[SIMSBIOHUB-519](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-519)

## Description of Changes

- added new migration to support SIMS holding event architecture
- migration based off of discussion surrounding this [miro board](https://miro.com/app/board/uXjVN5DXWE4=/?moveToWidget=3458764578433720586&cot=14)

## Testing Notes

- all migrations should run successfully
- subcount_event table and view should be removed

### CODE REVIEW

File: database/src/migrations/20240304143200_event_schema.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    DROP VIEW IF EXISTS biohub_dapi_v1.subcount_event;
+    DROP TABLE IF EXISTS subcount_event;
+
+    ----------------------------------------------------------------------------------------
+    -- Create measurement tables
+    ----------------------------------------------------------------------------------------
+    CREATE TABLE observation_subcount_quantitative_measurement (
+      observation_subcount_id INTEGER NOT NULL,
+      measurement_quantitative_id UUID NOT NULL,
+      value VARCHAR(100),
```
Comment (NickPhura) [inline database/src/migrations/20240304143200_event_schema.ts]: Can this just be `number`?

File: database/src/migrations/20240304143200_event_schema.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    ----------------------------------------------------------------------------------------
+    DROP VIEW IF EXISTS biohub_dapi_v1.subcount_event;
+    DROP TABLE IF EXISTS subcount_event;
+
+    ----------------------------------------------------------------------------------------
+    -- Create measurement tables
+    ----------------------------------------------------------------------------------------
+    CREATE TABLE observation_subcount_quantitative_measurement (
+      observation_subcount_id INTEGER NOT NULL,
+      measurement_quantitative_id UUID NOT NULL,
```
Comment (NickPhura) [inline database/src/migrations/20240304143200_event_schema.ts]: Forgot to include this before: to date, we've been prefixing external ids with the name of the system, so like `critterbase_measurement_quantitative_id`

**Comment by NickPhura** [comment]
@al-rosenthal Don't forget to add the views for the new tables

File: database/src/migrations/20240304143200_event_schema.ts (hunk 1 above)
Comment (al-rosenthal) [reply database/src/migrations/20240304143200_event_schema.ts]: we use integer in our other migrations

File: database/src/migrations/20240304143200_event_schema.ts (hunk 2 above)
Comment (al-rosenthal) [reply database/src/migrations/20240304143200_event_schema.ts]: updated

**Review by NickPhura — DISMISSED** [review]

**Review by NickPhura — APPROVED** [review]
🟦

**Review by KjartanE — APPROVED** [review]
🐈

**Pull request merged**